dts/bindings/gpio/quicklogic,eos-s3-gpio.yaml

```diff
@@ -0,0 +1,19 @@
+description: EOS S3 GPIO node
+
+compatible: "quicklogic,eos-s3-gpio"
+
+include: [gpio-controller.yaml, base.yaml]
+
+properties:
+    label:
+      required: true
+
+    interrupts:
+      required: true
+
+    "#gpio-cells":
+      const: 2
+
```

Comment:
This needs something like:
```
    ngpios:
      required: true
      const: 16
```
so the common code can determine the appropriate pin mask.  (Also this might help me understand whether the user-facing API provides 8 or 16 pins.)